src/certificate_provider.py

```diff
@@ -55,7 +55,7 @@ def request_certificate(self):
             del arguments['ServiceToken']
 
         if 'IdempotencyToken' not in arguments:
-            arguments['IdempotencyToken'] = self.request['LogicalResourceId']
+            arguments['IdempotencyToken'] = self.request['LogicalResourceId'] if len(self.request['LogicalResourceId']) <= 32 else self.request['LogicalResourceId'][:32]
```

Comment:
You could just write 
```python
arguments['IdempotencyToken'] = self.request['LogicalResourceId'][:32]
```

Reply:
Thanks @mvanholsteijn. We forked your repo and the push was meant to go there. Thanks for open sourcing, much appreciated 👍